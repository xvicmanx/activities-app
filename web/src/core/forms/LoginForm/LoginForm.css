.LoginForm {
  width: 90%;
  margin: auto;
  background: #fff;
  padding: 20px;
  border-radius: 5px;
}

.LoginForm__Title {
  text-align: center;
  font-size: 1.7rem;
}

.LoginForm .FormField {
  margin-bottom: 30px;
}

.LoginForm .FormField__label,
.LoginForm__Title {
  color: #9aa4aa;
}

.LoginForm .button {
  height: 2.5rem;
}

@media only screen and (min-width: 1024px) {
  .LoginForm {
    width: 70%;
  }
}