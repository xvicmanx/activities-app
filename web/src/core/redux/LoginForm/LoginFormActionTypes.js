export const HANDLE_CHANGE = 'HANDLE_CHANGE';
