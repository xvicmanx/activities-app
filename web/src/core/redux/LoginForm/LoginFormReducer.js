import { HANDLE_CHANGE } from './LoginFormActionTypes';

export const INITIAL_STATE = {
  email: '',
  password: '',
};

const LoginFormReducer = (state = INITIAL_STATE, { type, payload } = {}) => {
  switch (type) {
    case HANDLE_CHANGE:
      return {
        ...state,
        [payload.name]: payload.value,
      };

    default:
      return state;
  }
};

export default LoginFormReducer;
