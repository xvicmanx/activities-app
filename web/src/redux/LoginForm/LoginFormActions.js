import { HANDLE_CHANGE } from './LoginFormActionTypes';

export const handleChange = (e) => (dispatch) => {
  const { name, value } = e.target;

  dispatch({ type: HANDLE_CHANGE, payload: { name, value } });
};
