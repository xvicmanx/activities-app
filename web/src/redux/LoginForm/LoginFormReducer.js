import { HANDLE_CHANGE } from './LoginFormActionTypes';

const INITIAL_STATE = {
  email: '',
  password: ''
};

const LoginFormReducer = (state = INITIAL_STATE, { type, payload }) => {
  switch (type) {
    case HANDLE_CHANGE:
      return {
        ...state,
        [payload.name]: payload.value
      };

    default:
      return state;
  }
};

export default LoginFormReducer;
