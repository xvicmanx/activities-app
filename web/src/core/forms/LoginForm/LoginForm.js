import React from 'react';
import { Heading } from 'react-bulma-components';
import { useDispatch, useSelector } from 'react-redux';

import { handleChange } from '../../redux/LoginForm/LoginFormActions';
import { loginUser } from '../../../entities/users/redux/UsersActions';
import {
  Form,
  FormField,
  Button,
  Input,
  LoadingIndicator,
  ErrorMessage,
} from '../../../components';

import './LoginForm.css'; 


const LoginForm = () => {
  const { Users } = useSelector(state => state);
  const { email, password } = useSelector((state) => state.LoginForm);
  const dispatch = useDispatch();

  const handleOnChange = (e) => {
    dispatch(handleChange(e));
  };

  const handleOnSubmit = (e) => {
    e.preventDefault();
    dispatch(loginUser(email, password));
  };

  return (
    <div className="LoginForm">
      <Heading className="LoginForm__Title">
        Iniciar sesion
      </Heading>

      {Users.error && <ErrorMessage text={Users.error} />}

      <Form onSubmit={handleOnSubmit}>
        <FormField>
          <Input
            fullwidth="true"
            placeholder="Correo..."
            name="email"
            value={email}
            onChange={handleOnChange}
            autoComplete="off"
            type="email"
          />
        </FormField>

        <FormField>
          <Input
            fullwidth="true"
            placeholder="ContraseÃ±a..."
            name="password"
            value={password}
            onChange={handleOnChange}
            autoComplete="off"
            type="password"
          />
        </FormField>

        {Users.loading && <LoadingIndicator />}
        {!Users.loading && (
          <Button
            disabled={!password || !email}
            fullwidth
            type="submit"
            color="info"
          >
            Entrar
          </Button>
        )}
      </Form>
    </div>
  );
};

export default LoginForm;
