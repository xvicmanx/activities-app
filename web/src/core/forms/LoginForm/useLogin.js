// @flow

import { useDispatch, useSelector } from 'react-redux';
import { handleChange } from '../../redux/LoginForm/LoginFormActions';
import { loginUser } from '../../../entities/users/redux/UsersActions';

type Result = {
  loading?: boolean,
  email?: string,
  password?: string,
  onChange: Function,
  onSubmit: Function,
  error?: string,
};

const useLogin = (): Result => {
  const { Users } = useSelector((state) => state);
  const { email, password } = useSelector((state) => state.LoginForm);
  const dispatch = useDispatch();

  const onChange = (e) => {
    dispatch(handleChange(e));
  };
 
  const onSubmit = (e) => {
    e.preventDefault();
    dispatch(loginUser(email, password));
  };

 return {
    error: Users.error,
    loading: Users.loading,
    email,
    password,
    onChange,
    onSubmit,
  };
};

export default useLogin;
