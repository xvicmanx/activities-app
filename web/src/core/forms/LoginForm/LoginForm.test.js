import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import LoginForm from './LoginForm';

jest.mock('./useLogin');

import useLogin from './useLogin';

describe('LoginForm', () => {
  it('renders properly when not loading', () => {
    useLogin.mockImplementation(() => ({
      email: 'test-email',
      password: 'test-pass',
      onChange: () => {},
      onSubmit: () => {},
      loading: false,
      error: 'Error!',
    }));
    const renderer = new ShallowRenderer();
    const result = renderer.render(<LoginForm />);
    expect(result).toMatchSnapshot();
  });

  it('renders properly when loading', () => {
    useLogin.mockImplementation(() => ({
      email: 'test-email',
      password: 'test-pass',
      onChange: () => {},
      onSubmit: () => {},
      loading: true,
      error: null,
    }));
    const renderer = new ShallowRenderer();
    const result = renderer.render(<LoginForm />);
    expect(result).toMatchSnapshot();
  });
});
