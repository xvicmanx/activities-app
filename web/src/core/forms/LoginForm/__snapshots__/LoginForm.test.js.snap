// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders properly when loading 1`] = `
<div
  className="LoginForm"
>
  <Heading
    className="LoginForm__Title"
    heading={false}
    renderAs="h1"
    spaced={false}
    subtitle={false}
  >
    Iniciar sesion
  </Heading>
  <Form
    onSubmit={[Function]}
  >
    <FormField>
      <Input
        autoComplete="off"
        fullwidth="true"
        name="email"
        onChange={[Function]}
        placeholder="Correo..."
        type="email"
        value="test-email"
      />
    </FormField>
    <FormField>
      <Input
        autoComplete="off"
        fullwidth="true"
        name="password"
        onChange={[Function]}
        placeholder="Contraseña..."
        type="password"
        value="test-pass"
      />
    </FormField>
    <LoadingIndicator />
  </Form>
</div>
`;

exports[`LoginForm renders properly when not loading 1`] = `
<div
  className="LoginForm"
>
  <Heading
    className="LoginForm__Title"
    heading={false}
    renderAs="h1"
    spaced={false}
    subtitle={false}
  >
    Iniciar sesion
  </Heading>
  <ErrorMessage
    text="Error!"
  />
  <Form
    onSubmit={[Function]}
  >
    <FormField>
      <Input
        autoComplete="off"
        fullwidth="true"
        name="email"
        onChange={[Function]}
        placeholder="Correo..."
        type="email"
        value="test-email"
      />
    </FormField>
    <FormField>
      <Input
        autoComplete="off"
        fullwidth="true"
        name="password"
        onChange={[Function]}
        placeholder="Contraseña..."
        type="password"
        value="test-pass"
      />
    </FormField>
    <Button
      color="info"
      disabled={false}
      fullwidth={true}
      type="submit"
    >
      Entrar
    </Button>
  </Form>
</div>
`;
